### ISSUE TITLE
Also, how do I even have the OG role even though I only signed up yesterday.

### ISSUE THREAD

**Issue by riverscuomo** [riverscuomo/cuomputer#8]
ibonn

**Comment by DrGLDavidson** [comment]
In add_roles there is a default registered_on date 

        # A default for firestore user's who are missing registered on for some reason.
        registered_on = "Fri, 31 Jul 2020 00:00:00 GMT"

Which places that date as before the og_cutoff = datetime(2021, 9, 19)

So if a user's registeredOn is missing in the Firestore dictionary for whatever reason, they would be assigned the default and therefore OG. 

Could change registered_on to any date after September 19 2021 ? 

Maybe you want to do checks on your end to test my theory.
I noticed a new user joined 16 Feb 2025 and has the OG role.
username glopr display name emmaday
https://discord.com/channels/890210072381247548/895330177456963645/1341612817769562122

**Comment by riverscuomo** [comment]
@DrGLDavidson great catch! Can you make the change you suggested? Or is there a better way to handle user's who are missing registered on for some reason?

**Comment by DrGLDavidson** [comment]
Yup I can make the change. I can't think of a better way at the moment.

**Comment by riverscuomo** [comment]
@DrGLDavidson please let me know if there is now an outstanding issue of OG role that needs to be removed from some users.

**Comment by DrGLDavidson** [comment]
@riverscuomo yes there is. 
ibonn is still OG but shouldn't be. gplor could be a genuine OG who just (re)joined discord, and their original registeredOn was retained. I don't know how many other users have an erroneous OG role. 

The 25 or so users who were missing registeredOn  you said you fixed on #75, did they have no date at all or the default registered_on?
I ask because one option could be to use the default 31 July 2020 date as a filter to remove erroneous OGs (assuming no genuine OG also signed up at midnight on the 31 July 2020).

**Comment by riverscuomo** [comment]
They had no `registeredOn` property at all. I gave theme the date one week before their `lastSeen` property.

Now I need to write a script to remove "OG" badge for anyone whose  `registeredOn` is after the OG cutoff date.

**Comment by DrGLDavidson** [comment]
keep issue open? ibonn still has OG
